patches/components-history-core-browser-sync-typed_url_sync_bridge.cc.patch

```diff
@@ -0,0 +1,44 @@
+diff --git a/components/history/core/browser/sync/typed_url_sync_bridge.cc b/components/history/core/browser/sync/typed_url_sync_bridge.cc
+index 487d267c3a423c613fd41c74893933355f1d2a97..155bac5fee7111cddfba202c30d66008c26e0739 100644
+--- a/components/history/core/browser/sync/typed_url_sync_bridge.cc
++++ b/components/history/core/browser/sync/typed_url_sync_bridge.cc
+@@ -68,6 +68,7 @@ std::string GetStorageKeyFromURLRow(const URLRow& row) {
+ }
+ 
+ bool HasTypedUrl(const std::vector<VisitRow>& visits) {
++  BRAVE_HAS_TYPED_URL_RETURN_TRUE_IF_SEND_ALL_HISTORY_ENABLED
+   return base::ranges::any_of(visits, [](const VisitRow& visit) {
+     return ui::PageTransitionCoreTypeIs(visit.transition,
+                                         ui::PAGE_TRANSITION_TYPED);
+@@ -239,6 +240,7 @@ absl::optional<syncer::ModelError> TypedURLSyncBridge::ApplySyncChanges(
+       continue;
+     }
+ 
++    BRAVE_TYPED_URL_SYNC_BRIDGE_APPLY_SYNC_CHANGES_SKIP_DCHECK
+     DCHECK(specifics.visits_size());
+     sync_pb::TypedUrlSpecifics filtered_url = FilterExpiredVisits(specifics);
+     if (filtered_url.visits_size() == 0) {
+@@ -527,6 +529,7 @@ bool TypedURLSyncBridge::WriteToTypedUrlSpecifics(
+     // Walk the passed-in visit vector and count the # of typed visits.
+     for (const VisitRow& visit : visits) {
+       // We ignore reload visits.
++      BRAVE_WRITE_TO_TYPED_URL_SPECIFICS_DONT_IGNORE_RELOADS
+       if (PageTransitionCoreTypeIs(visit.transition,
+                                    ui::PAGE_TRANSITION_RELOAD)) {
+         continue;
+@@ -554,6 +557,7 @@ bool TypedURLSyncBridge::WriteToTypedUrlSpecifics(
+ 
+   for (const VisitRow& visit : visits) {
+     // Skip reload visits.
++    BRAVE_WRITE_TO_TYPED_URL_SPECIFICS_DONT_IGNORE_RELOADS
+     if (PageTransitionCoreTypeIs(visit.transition,
+                                  ui::PAGE_TRANSITION_RELOAD)) {
+       continue;
+@@ -1136,6 +1140,7 @@ bool TypedURLSyncBridge::ShouldIgnoreVisits(
+ 
+ bool TypedURLSyncBridge::ShouldSyncVisit(int typed_count,
+                                          ui::PageTransition transition) {
++  BRAVE_TYPED_URL_SYNC_BRIDGE_SHOULD_SYNC_VISIT_RETURN_TRUE_IF_SEND_ALL_HISTORY_ENABLED
```

Comment:
maybe it's possible to subclass something here? the PR is fine, but the amount of defines we put into this class is quite notable.

Reply:
Thanks. I will try to subclass to reduce the patches.

I considered to redefine function names, but it didn't work, because it is private and used in the same file where it's body is.

Reply:
shouldn't we keep the `kTypedUrlVisitThrottleThreshold` and `kTypedUrlVisitThrottleMultiple` check for TypedUrl here? And I think we should impose some threshold for other types too if we want to sync them all. Maybe sync for the first time (`typed_count == 1`) and same threshold 10 and 10 multiple for `type_count > 1`?

Reply:
@goodov 
I have done the changes to subclass TypedURLSyncBridge 
https://github.com/brave/brave-core/pull/16705/commits/1262573f61b3753ed52bf1531675b885767dedd5

I was able to get rid of this patch line https://github.com/brave/brave-core/pull/16705/commits/1262573f61b3753ed52bf1531675b885767dedd5#diff-6243db085330c064cecdd1d0782dd214a2e1822a36b23d3847555a8425b49c98L41

but I had to introduce this general patch to BUILD.gn at 
https://github.com/brave/brave-core/pull/16705/commits/1262573f61b3753ed52bf1531675b885767dedd5#diff-4b1a1eea4b4ca7407c1f0444e4f12750c88b469a453b57b622c673012e99b5d0R9

@darkdh 
Thea answer about `kTypedUrlVisitThrottleThreshold` and `kTypedUrlVisitThrottleMultiple` is on the way, but I feel like you are right.

@goodov and @darkdh 
From what I have seen today, `chromium_src` override doesn't work for *.cc for unit tests. Maybe this is well-know issue, but I had to explicitly change ["//components/history/core/browser/sync/typed_url_sync_bridge_unittest.cc" => "//brave/chromium_src/components/history/core/browser/sync/typed_url_sync_bridge_unittest.cc"](https://github.com/brave/brave-core/pull/16705/commits/1262573f61b3753ed52bf1531675b885767dedd5#diff-02a56120c401438355c06b792cc494b41c2440cab96b107ce8fc8a49353b22f4R109)
because otherwise [this change of `#define TypedURLSyncBridge BraveTypedURLSyncBridge`](https://github.com/brave/brave-core/pull/16705/commits/1262573f61b3753ed52bf1531675b885767dedd5#diff-c44b2804c93e3485c6d2c63e4d694a05716cddc129bd7823ebc8dc382c082323R21) wasn't picked up.

I found that with `"//components/history/core/browser/sync/typed_url_sync_bridge_unittest.cc"` BUILD.gn  entry the compiler actually compiled indeed `"//components/history/core/browser/sync/typed_url_sync_bridge_unittest.cc"` instead of `"//brave/chromium_src/components/history/core/browser/sync/typed_url_sync_bridge_unittest.cc",` so I had to  change this.

Reply:
UPDATE: The statement about _chromium_src override doesn't work for *.cc for unit tests_ from above is wrong.